File: tobas_core/tobas_rqt_tools/src/tobas_rqt_tools/widgets/widget.py
import rclpy
from overrides import override
from PyQt5.QtWidgets import QWidget


class Widget(QWidget):
    """
    ===== QWidgetとの違い =====
    - closeで子ウィジェットのcloseを再帰的に呼び出す
    """

    @override
    def close(self) -> bool:
        rclpy.logdebug(f"{self.__class__.__name__}.close")

        for child in self.findChildren(Widget):
            child.close()

        return super().close()

File: tobas_gui/tobas_qt_tools/src/widgets/widget.cpp
#include "tobas_qt_tools/widgets/widget.hpp"

namespace qt
{
bool Widget::close()
{
  for (const auto child : findChildren<Widget*>())
    child->close();

  return super::close();
}

QPoint Widget::getCenter() const
{
  return QPoint(width() / 2, height() / 2);
}
}  // namespace qt
